docs/core/diagnostics/metrics-collection.md

```diff
@@ -0,0 +1,404 @@
+                })
+                .Build();
+
+        Console.WriteLine("Press any key to exit");
+        while(!Console.KeyAvailable)
+        {
+            // Pretend our store has a transaction each second that sells 4 hats
+            Thread.Sleep(1000);
+            s_hatsSold.Add(4);
+        }
+    }
+}
+```
+
+`AddMeter("HatCo.HatStore")` configures OpenTelemetry to transmit all the metrics collected by the Meter our app defined.
+`AddPrometheusExporter(...)` configures OpenTelemetry to expose Prometheus' metrics endpoint on port 9184 and to use
+the HttpListener. See the [OpenTelemetry documentation](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Exporter.Prometheus)
+for more information about OpenTelemetry configuration options, in particular alternate hosting options that are useful for ASP.NET applications.
+
+> [!NOTE]
+> At the time of writing OpenTelemetry only supports metrics emitted using the <xref:System.Diagnostics.Metrics?displayProperty=nameWithType>
+> APIs however support for [EventCounters](event-counters.md) is planned.
+
+3. Run the sample app and leave it running in the background.
+
+```dotnetcli
+> dotnet run
+Press any key to exit
+```
+
+#### Setup and Configure Prometheus
+
+1. Follow the [Prometheus first steps](https://prometheus.io/docs/introduction/first_steps/) to set up your Prometheus server
+and confirm it is working.
+
+2. Modify the prometheus.yml configuration file so that Prometheus will scrape the metrics endpoint that our example app is
+exposing. Add this text in the scrape_configs section:
+
+```
+  - job_name: 'OpenTelemetryTest'
+    static_configs:
+    - targets: ['localhost:9184']
+```
+
+If you are starting from the default configuration then scrape_configs should now look like this:
+
+```
+scrape_configs:
+  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
+  - job_name: "prometheus"
+
+    # metrics_path defaults to '/metrics'
+    # scheme defaults to 'http'.
+
+    static_configs:
+      - targets: ["localhost:9090"]
+
+  - job_name: 'OpenTelemetryTest'
+    static_configs:
+      - targets: ['localhost:9184']
+```
+
+Reload the configuration or restart the prometheus server, then confirm that OpenTelemetryTest is in the UP
+state in the Status->Targets page of the Prometheus web portal.
+
+3. On the Graph page of the Prometheus web portal enter `hats_sold` in the expression text box. In the graph tab Prometheus should
+show the steadily increasing value of the "hats-sold" Counter that is being emitted by our sample application.
+[hats-sold-graph](media\prometheus-hat-sold-metric.png)
+
+If the Prometheus server hasn't been scraping (Prometheus' terminology for polling) the sample app for long you may need to wait a short
+while for data to accumulate. The default configuration has Prometheus scrape metric data every 15 seconds.
```

Comment:
I suggest putting an explicit scraping period in the configuration file (e.g. 1 second) for better demo/learning experience.